src/core/stream/representation/representation_stream.ts

```diff
@@ -481,6 +485,34 @@ export default function RepresentationStream<T>({
 
       case "parsed-segment":
         const initSegmentData = initSegmentObject?.initializationData ?? null;
+        if (evt.value.emsgs !== undefined) {
+          const { needsManifestRefresh, nonInterpretedMessages } = evt.value.emsgs
+            .reduce((acc, val: IEventMessage) => {
+              // Scheme that signals manifest update
+              if (val.schemeId === "urn:mpeg:dash:event:2012") {
+                acc.needsManifestRefresh = true;
```

Comment:
:O Even if the segment is not close at all to the live edge?

Reply:
I didn't read DASH-IF there, just asking.

Reply:
In DASH-if v4.3, it's explained in
> 4.5.2. Service Offering Requirements and Guidelines

It's actually more complicated than that

Reply:
Yes, i saw that part of DASH-if. I'm going to study it.